### PR TITLE
Add AVX2 support for  Windows

### PR DESCRIPTION
**Pull request by robbmcleod**
- Enables AVX2 in the blosc macros for Windows builds.
- Updates `cpuinfo.py` to version `4.0.0`.
- Protects entire `setup.py` with a `if __name__ == '__main__'` block to avoid `cpuinfo` issue related to multiprocessing: https://github.com/workhorsy/py-cpuinfo/issues/108

### CODE REVIEW

**Comment by FrancescAlted** [comment]
This looks like a pretty big change, but interesting anyway.  I think I should wait a bit and see if that can be fixed in upstream cpuinfo, but if not, I am +1 on merging this.

Thanks!

**Comment by FrancescAlted** [comment]
Ok, we have waited enough.  Merging.

File: setup.py [hunk 1]
```diff
@@ -31,1024 +32,1034 @@
...
+        # Finally, add all the include dirs...
+        inc_dirs += [os.path.join('c-blosc', 'blosc')]
+        inc_dirs += glob.glob('c-blosc/internal-complibs/*')
+        inc_dirs += glob.glob('c-blosc/internal-complibs/zstd*/common')
+        inc_dirs += glob.glob('c-blosc/internal-complibs/zstd*')
+        # ...and the macros for all the compressors supported
+        def_macros += [('HAVE_LZ4', 1), ('HAVE_SNAPPY', 1), ('HAVE_ZLIB', 1),
+                    ('HAVE_ZSTD', 1)]
+
+        #
```
Comment (FrancescAlted) [inline setup.py]: @robbmcleod After inspecting this more carefully, shouldn't we add the avx2 source files to the windows ('nt') section above?

Comment (robbmcleod) [reply setup.py]: This isn't fresh in my mind anymore, but isn't the code fragment from lines 885 to 896 doing  that?

```suggestion
             if 'avx2' in cpu_flags:
            print('AVX2 detected')
            if os.name == 'nt':
                if LooseVersion(platform.python_version()) >= LooseVersion('3.5.0'):
                    # Neither MSVC2008 for Python 2.7 or MSVC2010 for Python 3.4 have 
                    # sufficient AVX2 support
                    def_macros += [('__AVX2__', 1)]
            elif compiler_has_flags(compiler, ["-mavx2"]):
                CFLAGS.append('-DSHUFFLE_AVX2_ENABLED')
                CFLAGS.append('-mavx2')
                blosc_sources += [f for f in glob.glob('c-blosc/blosc/*.c')
                                if 'avx2' in f]
```

Comment (FrancescAlted) [reply setup.py]: Yeah, but not for `os.name == 'nt'`, so unless I am missing something, the AVX2 support is not reaching to windows.  My current solution (already committed) is to deindent the iclusion of 'avx2' files one level:

```
        if 'avx2' in cpu_flags:
            print('AVX2 detected')
            if os.name == 'nt':
                if LooseVersion(platform.python_version()) >= LooseVersion('3.5.0'):
                    # Neither MSVC2008 for Python 2.7 or MSVC2010 for Python 3.4 have 
                    # sufficient AVX2 support
                    def_macros += [('__AVX2__', 1)]
            elif compiler_has_flags(compiler, ["-mavx2"]):
                CFLAGS.append('-DSHUFFLE_AVX2_ENABLED')
                CFLAGS.append('-mavx2')
            blosc_sources += [f for f in glob.glob('c-blosc/blosc/*.c')  if 'avx2' in f]
```

Makes sense?

Comment (FrancescAlted) [reply setup.py]: Oops, correction, the avx2 sources should be included only if the conditions are fulfilled, so better like this:

```
        if 'avx2' in cpu_flags:
            print('AVX2 detected')
            if os.name == 'nt':
                if LooseVersion(platform.python_version()) >= LooseVersion('3.5.0'):
                    # Neither MSVC2008 for Python 2.7 or MSVC2010 for Python 3.4 have 
                    # sufficient AVX2 support
                    def_macros += [('__AVX2__', 1)]
                    blosc_sources += [f for f in glob.glob('c-blosc/blosc/*.c')  if 'avx2' in f]
            elif compiler_has_flags(compiler, ["-mavx2"]):
                CFLAGS.append('-DSHUFFLE_AVX2_ENABLED')
                CFLAGS.append('-mavx2')
                blosc_sources += [f for f in glob.glob('c-blosc/blosc/*.c')  if 'avx2' in f]
```

Agreed?

Comment (robbmcleod) [reply setup.py]: Agreed.

**Pull request merged**